src/clj/puppetlabs/enterprise/services/puppet_profiler/puppet_profiler_core.clj

```diff
@@ -0,0 +1,132 @@
+(ns puppetlabs.enterprise.services.puppet-profiler.puppet-profiler-core
+  (:import (com.codahale.metrics MetricRegistry Timer)
+           (com.puppetlabs.enterprise MetricsPuppetProfiler)
+           (com.puppetlabs.puppetserver PuppetProfiler))
+  (:require [clojure.string :as str]
+            [schema.core :as schema]
+            [puppetlabs.kitchensink.core :as ks]
+            [clojure.tools.logging :as log]
+            [puppetlabs.trapperkeeper.services.status.status-core :as status-core]
+            [puppetlabs.metrics :as metrics]
+            [puppetlabs.i18n.core :refer [trs]]))
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Schemas
+
+(def MetricsProfilerServiceContext
+  {:profiler (schema/maybe PuppetProfiler)})
+
+(def MetricsProfilerServiceConfig
+  {(schema/optional-key :enabled) schema/Bool})
+
+(def PEPuppetProfilerStatusV1
+  {(schema/optional-key :experimental)
+   {:function-metrics [{:function schema/Str
+                        :count schema/Int
+                        :mean schema/Num
+                        :aggregate schema/Num}]
+    :resource-metrics [{:resource schema/Str
+                        :count schema/Int
+                        :mean schema/Num
+                        :aggregate schema/Num}]
+    :catalog-metrics [{:metric schema/Str
+                       :count schema/Int
+                       :mean schema/Num
+                       :aggregate schema/Num}]
+    :inline-metrics [{:metric schema/Str
+                      :count schema/Int}]}})
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; Private
+
+(schema/defn metrics-profiler :- PuppetProfiler
```

Comment:
Newb question: since these functions are private can we use `defn-` to limit their accessibility outside of this namespace? (And would that work for functions with schemas?)